### PR TITLE
add sphinx reset and small refactor 

### PR DESCRIPTION
**Pull request by akostadinov**

### CODE REVIEW

**Review by RobGri001 — APPROVED** [review]

File: bin/porta [hunk 1]
```diff
@@ -626,10 +688,26 @@ module Porta
...
+    def reset_redis
       system('redis-cli flushall') || raise("redis-cli execution failed")
-      in_porta(:system, porta_additional_envs) { "bundle exec rails db:reset" } || raise("db:reset in Porta failed")
-      in_zync {"bundle exec rails db:reset" } || raise("db:reset in Zync failed")
+    end
+
+    def reset_system
+      run_apisonator_listener && in_porta(:system, porta_additional_envs) { "bundle exec rails db:reset" } || raise("db:reset in Porta failed")
+    ensure
+      stop_apisonator_listener
```
Comment (jlledom) [inline bin/porta]: what if it was already working before? this will kill it and force the user to manually launch it after. Or am I wrong?

Comment (akostadinov) [reply bin/porta]: yes, but resetting databases while deps are running can confuse them anyway. So its not a big deal IMO.

But I'll try to make this work with the fake server by updating `db:seed` task in porta. I think it makes sense to use fake_server always while seeding the DB as expecting backend to be clean is not guaranteed. And we have `resync` in the instructions already. Let me know what you think.

Comment (jlledom) [reply bin/porta]: I don't understand why `porta reset` needs apisonator to be up. If it can be replaced by the fake server, that's better because it'll remove a dependency.

On the other hand, it usually happens to me that I run `porta reset` and the `db:seed` step fails because some of the deps are up or down when they shouldn't. For instance, I usually see an error complaining about zync already running. It's not clear for me what should and shouldn't be running when in launch `porta reset`. I think the dependencies for this command should be either enforced in the code or documented in the its help dialog.

Comment (akostadinov) [reply bin/porta]: Steps are documented in the porta-dev-tools readme at the bottom. Enforcing order of commands is outside the scope of this PR.

When you configure fake server, then the changes are not reflected in the real apisonator so API calls wont work. That's why changed porta config to use real apisonator that porta-dev-tools starts. But issue arises on reset because porta tries to contact apisonator during db seed. But apisonator should be down for the reset or redis. This is about making things work simple out of the box.

if apisonator is running during db:seed, then it will pick-up the changes and `resync` will not be needed later. So in general doing what this PR does is more natural than using fake server. But it is a little complicated so I'm considering to always use fake server enforced in porta code, during `db:seed` and then user will need to `resync` backend.

More clear?

Comment (jlledom) [reply bin/porta]: yes, more clear. Why not then calling also `stop_apisonator_listener` from `reset_redis` that way apisonator would be down for redis and up for system.

Comment (akostadinov) [reply bin/porta]: as I said, protecting users from error is not in the scope. It is not only apisonator. Also sidekiq and porta need to be down for redis. zync and zync queue must be down for postgres.

But also, I want to remove this once I can come to it again. I think before we make it smarter, we need to think it through more on the high level.

Comment (akostadinov) [reply bin/porta]: This fixes the issue on porta side: 3scale/porta#3283

Also will not hurt any use case we want to implement here.

Comment (jlledom) [reply bin/porta]: Then, it's this still required?

Comment (akostadinov) [reply bin/porta]: Not required, only "if we want". For example to avoid the `resync` need.

Comment (jlledom) [reply bin/porta]: Then the `README` should be updated too

File: bin/porta [hunk 2]
```diff
@@ -684,41 +771,6 @@ module Porta
       run_apisonator && run_porxy && run_apicast && run_sphinx && run_zync
     end
```
Comment (jlledom) [inline bin/porta]: We don't call `run_deps` anymore, right? We could remove it.

**Review by jlledom — COMMENTED** [review]
The PR title doesn't match the code changes, this is more a code refactor.

Which was the goal, to sync the backend after `db:reset`? Is that achieved now by setting `CONFIG_INTERNAL_API_USER` and `CONFIG_INTERNAL_API_PASSSWORD`? If that's the case, maybe we should create a PR only for that. Otherwise the git log is going to be hard to follow.

Also, when I check out this and run `porta deps` everything works except apicast. The container fails with this output:

`nginx: [emerg] invalid IPv6 address in resolver "[fe80::1%2]:53" in /tmp/lua_ltHeOf:52`

File: bin/porta (hunk 2 above)
Comment (akostadinov) [reply bin/porta]: what do you mean with we don't call `run_deps`?

**Comment by akostadinov** [comment]
> Also, when I check out this and run porta deps everything works except apicast.

This PR doesn't change anything with apicast. Can you check if without PR it does (not) work?

File: bin/porta (hunk 2 above)
Comment (jlledom) [reply bin/porta]: `run_deps` was being called from here:

https://github.com/3scale-labs/porta-dev-tools/pull/18/files#diff-422e0549dec116dd1a61191fce586358e3ec8f56ebbe702d29b738ffdbe22a5cL671

Now nobody calls it, afaik.

**Comment by jlledom** [comment]
> > Also, when I check out this and run porta deps everything works except apicast.
> 
> This PR doesn't change anything with apicast. Can you check if without PR it does (not) work?

Yes, I checked it, it works on master and fails in this branch, the error is about IPv6 
¯\\_(ツ)_/¯

File: bin/porta (hunk 2 above)
Comment (akostadinov) [reply bin/porta]: got it , will remove, good catch!

**Comment by akostadinov** [comment]
@jlledom , can you check what command is being used to run apicast with this branch compared to master. For me it uses exactly the same command both ways:
```sh
docker run -d --name apicast --rm -p 8080:8080 --add-host=host.docker.internal:10.0.2.2 --net slirp4netns:allow_host_loopback=true,enable_ipv6=false -e THREESCALE_PORTAL_ENDPOINT="http://apicastsecret@10.0.2.2:3008/master/api/proxy/configs" -e THREESCALE_DEPLOYMENT_ENV=staging -e BACKEND_ENDPOINT_OVERRIDE="http://10.0.2.2:3001" quay.io/3scale/apicast:latest >/dev/null && echo "apicast"
```

You can check it by applying such patch:
```diff
diff --git a/bin/porta b/bin/porta
index 3fae2cf..379ad20 100755
--- a/bin/porta
+++ b/bin/porta
@@ -724,6 +724,7 @@ module Porta
     end
 
     def run_apicast
+      require 'pry'; binding.pry
       system("docker run -d --name apicast --rm -p 8080:8080 #{container_opts_local_hosts} -e THREESCALE_PORTAL_ENDPOINT=\"http://#{options[:apicast_access_token]}@#{docker_internal_host}:3008/master/api/proxy/configs\" -e THREESCALE_DEPLOYMENT_ENV=staging -e BACKEND_ENDPOINT_OVERRIDE=\"http://#{docker_internal_host}:3001\" #{options[:apicast_image]} >/dev/null && echo \"apicast\"")
     end
```

I don't see any changes related to that in this branch but I assume the startup command will show.

**Comment by jlledom** [comment]
> can you check what command is being used to run apicastç

I checked and it's exactly the same command. But now it doesn't fail... so never mind.

File: bin/porta (hunk 2 above)
Comment (akostadinov) [reply bin/porta]: removed!

**Review by jlledom — APPROVED** [review]
